rpc/src/commands/chains/blocks/get_blocks_in_chain.rs

```diff
@@ -1,16 +1,88 @@
+use super::block_responses::BlocksInChainResponse;
 use crate::commands::RpcClientCommand;
-use crate::responses::chains::blocks::block_ids_in_chain::BlocksInChainResponse;
 use crate::types::Chain;
+use chrono::NaiveDateTime;
+use querystring;
 
+type BlockHash = String;
+
+#[derive(Debug)]
+/// Command for the [`/chains/{chain_id}/blocks` endpoint](https://tezos.gitlab.io/shell/rpc.html#get-chains-chain-id-blocks).
+///
+/// Requires a valid [`chain_id`](Chain) to form the URL string
+///
+/// Can optionally hold extra query parameters with
+/// [an explicit constructor](Self::with_explicit_params),
+/// or default to sending no args with [the default constructor](Self::with_default_params)
 pub struct GetBlocksInChain {
     pub chain_id: Chain,
+    params: Option<GetBlocksInChainParameters>,
+}
+
+#[derive(Debug)]
+struct GetBlocksInChainParameters {
```

Comment:
Alright, this might be super extra but we can definitely use a macro for parameter structs so we can easily turn them into a vector of tuples. [Check it out](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=d8712a938239006da0ce5bf8bfb3ef09)

It doesn't use the `querystring` crate since I couldn't import it but that's a small change. Also as you can see it only calls `to_string()` and doesn't handle special cases like `timestamp().to_string()` but since the RPC spec expects a timestamp the SDK users should be providing one anyways instead of us handling it

Reply:
That macro is very very sexy, but i dont think `Vec<(String, String)>` is needed in a url/query formatting context. The only reason I can think of in which we would need a string tuple is maybe for comparison, but at that point it would be easier to just compare the actual query objects imo.

Willing to bike shed on this topic because i think its a real problem, albeit one we've mostly implemented a solution for with `querystring`. Thoughts? @liam0215

Reply:
Yeah, I agree the `Vec<(String, String)>` isn't needed but in reality, it would return a `String`. The only reason I had it return a vector of string tuples was that I couldn't use the `querystring` crate, if added we would take that vector of tuples and pass it into `querystring::stringify` so the fn returns the URL query string. With this, we can create query param structs at free will and let the macro *kinda* act as a trait

Reply:
ahhh I see, i totally missed the missing import comment (my bad!).
With that being said, I think this is an extremely elegant solution, especially long-term. 

I'll work on implementing this out, but holy moly is this good code. Kudos.